internal/metamorph/zmq.go

```diff
@@ -56,21 +77,26 @@ type CollidingTx struct {
 	Size int    `json:"size"`
 }
 
-func NewZMQ(zmqURL *url.URL, statusMessageCh chan<- *PeerTxMessage, logger *slog.Logger) *ZMQ {
+type ZMQI interface {
+	Subscribe(string, chan []string) error
+}
+
+func NewZMQ(ctx context.Context, zmqURL *url.URL, statusMessageCh chan<- *PeerTxMessage, zmqHandler ZMQI, logger *slog.Logger) *ZMQ {
+	if zmqHandler == nil {
+		zmqHandler = NewZMQHandler(ctx, zmqURL, logger)
```

Comment:
If the zmq handler is `nil`, I think we should return an error here

Reply:
Done